pandas/tests/arithmetic/test_period.py

```diff
@@ -1276,13 +1276,15 @@ def test_pi_ops(self):
 
         self._check(idx + 2, lambda x: x - 2, idx)
 
-        result = idx - Period("2011-01", freq="M")
-        off = idx.freq
-        exp = pd.Index([0 * off, 1 * off, 2 * off, 3 * off], name="idx")
+        with tm.assert_produces_warning(FutureWarning, check_stacklevel=False):
+            result = idx - Period("2011-01", freq="M")
+            off = idx.freq
+            exp = pd.Index([0 * off, 1 * off, 2 * off, 3 * off], name="idx")
```

Comment:
once this deprecation is enforced, what would happen here?  just not-allowing subtraction seems undesirable

Reply:
Yes, true that it seems undesirable. Any suggestions? I thought of returning an offset that allows for `n=0`, but that could be undesirable if other time arithmetic operations are involved.

For reference, when we get here, we have
```python
ipdb> idx                                                                                                                                     
PeriodIndex(['2011-01', '2011-02', '2011-03', '2011-04'], dtype='period[M]', name='idx', freq='M')
```

Reply:
Looking back at #31184 wouldnt removing the incorrect doc also be a viable option?

Reply:
LOL yeah, that would be a lot simpler :) Perhaps there could still be a warning that an offset with `n=0` has been created - what do you think?

Reply:
could the warning go closer to the problematic behavior?